Artesian/Artesian.SDK/Dto/Acl/AuthorizationPrincipalRole.cs

```diff
@@ -24,19 +24,20 @@ public record Principal
         /// The Principal Type
         /// </summary>
         [Key(0)]
-        public PrincipalType PrincipalType { get; set; }
+        public PrincipalType PrincipalType { get; init; }
+        
         /// <summary>
         /// The Principal Identifier
         /// </summary>
         [Key(1)]
-        public string PrincipalId { get; set; }
+        public required string PrincipalId { get; init; }
 
         /// <summary>
-        /// 
+        /// Default constructor for serialization
         /// </summary>
         public Principal()
```

Comment:
remove both ctor from this record
move the logic of the Private ctor into the converter operator